stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -0,0 +1,404 @@
+        "cases, this list can be very long. That's why this option is unchecked"
+        "by default.",
+    )
+
+    __slots__ = (
+        "_volume_by_partners",
+        "_device",
+        "_remaining_weight",
+        "_remaining_nbr_picking_lines",
+        "_selected_picking_ids",
+        "_first_picking",
+        "_remaining_nbr_bins",
+        "_previous_selected_picking",
+    )
+
+    def __init__(self, env, ids=(), prefetch_ids=()):
+        super().__init__(env, ids=ids, prefetch_ids=prefetch_ids)
+        self._reset_counters()
+
+    def create_batch(self):
+        self.ensure_one()
+        batch = self._create_batch(raise_if_not_possible=True)
+        action = {
+            "type": "ir.actions.act_window",
+            "name": batch.name,
+            "res_model": "stock.picking.batch",
+            "res_id": batch.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+        return action
+
+    def _reset_counters(self):
+        self._volume_by_partners = defaultdict(lambda: 0)
+        self._device = None
+        self._remaining_weight = 0
+        self._remaining_nbr_picking_lines = 0
+        self._selected_picking_ids = []
+        self._first_picking = None
+        self._remaining_nbr_bins = None
+        self._previous_selected_picking = None
+
+    def _get_picking_order_by(self):
+        self.ensure_one()
+        if self.group_pickings_by_partner:
+            return "priority desc, scheduled_date asc, partner_id desc, id asc"
+        return "priority desc, scheduled_date asc, id asc"
+
+    def _get_picking_domain_common(self, user=None):
+        domain = [
+            ("picking_type_id", "in", self.picking_type_ids.ids),
+            ("state", "in", ("partially_available", "assigned")),
+            ("batch_id", "=", False),
+            ("user_id", "=", user.id if user else False),
+        ]
+        if not user:
+            domain.append(("printed", "=", False))
+        return domain
+
+    def _get_picking_domain_for_device(self, device):
+        return [
+            ("volume", ">=", device.min_volume),
+            ("volume", "<=", device.max_volume),
+            ("weight", "<=", device.max_weight),
+            ("nbr_picking_lines", "<=", self.maximum_number_of_preparation_lines),
+        ]
+
+    def _get_picking_domain_for_additional(self):
+        """Provides the domain expressing the additional constraints to apply to
+        get the next picking to add to the batch.
+        """
+        excluded_ids = self._selected_picking_ids
+        domain = [
+            (
+                "nbr_picking_lines",
+                "<=",
+                self._remaining_nbr_picking_lines,
+            ),
+            ("id", "not in", excluded_ids),
+            ("weight", "<=", self._remaining_weight),
+        ]
+        user = self._first_picking.user_id
+        if user:
+            domain.append(("user_id", "=", user.id))
```

Comment:
Should be like above. If there is no user_id on the first picking, you cannot steal from another user
```suggestion
            domain.append(("user_id", "in", [user.id, False] if self.user_id or [False]))
```